src/porepy/viz/solver_statistics.py

```diff
@@ -0,0 +1,91 @@
+"""Solver statistics object for non-linear solver loop."""
+
+from __future__ import annotations
+
+import json
+import logging
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Optional
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class SolverStatistics:
+    """Statistics object for non-linear solver loop.
+
+    This object keeps track of the number of non-linear iterations performed for the
+    current time step, as well as increments and residuals for each iteration.
+
+    Example:
+
+        After storing solver statistics to file, we can load the file and plot
+        analogous data, here for the first time step.
+
+        >>> import matplotlib.pyplot as plt
+        >>> import json
+        >>> with open("solver_statistics.json", "r") as f:
+        >>>     history = json.load(f)
+        >>> ts = str(1)
+        >>> err = history[ts]["residual_errors"]
+        >>> plt.semilogy(err)
+        >>> plt.xlabel("Iteration number")
+        >>> plt.ylabel("Residual")
+        >>> plt.title("Residual error")
+        >>> plt.show()
+
+    """
+
+    num_iteration: int = 0
+    """Number of non-linear iterations performed for current time step."""
+    increment_errors: list[float] = field(default_factory=list)
+    """List of increments for each non-linear iteration."""
+    residual_errors: list[float] = field(default_factory=list)
+    """List of residual for each non-linear iteration."""
+    path: Optional[Path] = None
+    """Path to save the statistics object to."""
+
+    def log_error(self, increment_error: float, residual_error: float) -> None:
```

Comment:
If I want to keep track of other errors, I'd have to change the signature of this method, right? Is it worth considering to replace the explicit arguments by kwargs?

Reply:
Or a combination. Always storing the main errors seem sound to me.

Reply:
If you would want to customize what errors are kept track of/stored to file, more changes would be needed at the moment. So my answer atm would be "no", but in the near future, I may change my view.

Reply:
Sorry, was working on outdated window and did not see comments by @keileg. The main idea here was just to store the main errors for now. The functionality for storing other errors has been removed for this PR.

Reply:
Fair enough

Reply:
Yes, but we should avoid changing signatures in two months.

Reply:
@keileg @IvarStefansson I have no strong opinion here. I would keep the signature, as I think the standard errors will remain no mater how new functionality will enter. And optional errors will then be added through kwargs. Of course one could also consider using kwargs for all of them. If you want to decide, give me a sign and I will update the signature here accordingly. This is the last open comment.

Reply:
Then let us add `kwargs` as a way to include future extensions, so that the new signature is 

```python
    def log_error(self, increment_error: float, residual_error: float, **kwargs) -> None:
```

Reply:
Done.